test/mock/MockMulticall.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+import "../../contracts/base/Multicall.sol";
+
+contract MockMulticall is Multicall {
+    function functionThatRevertsWithError(string memory error) external pure {
+        revert(error);
+    }
+
+    struct Tuple {
+        uint256 a;
+        uint256 b;
+    }
+
+    function functionThatReturnsTuple(uint256 a, uint256 b) external pure returns (Tuple memory tuple) {
+        tuple = Tuple({a: a, b: b});
+    }
+
+    uint256 public paid;
+
+    function pays() external payable {
+        paid += msg.value;
```

Comment:
if you call pays twice in the multicall wont it set `paid` to msg.value *2, but you've actually only sent in msg.value ?